elements/pf-date-picker/pf-calendar.ts

```diff
@@ -0,0 +1,269 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { ref, createRef } from 'lit/directives/ref.js';
+import styles from './pf-calendar.css';
+import { getFormattedDate, getMonthNamesFromLocale } from './date-picker-helper.js';
+import { ComposedEvent } from '@patternfly/pfe-core/core.js';
+
+export interface FocusedDateValues{
+  day: number;
+  month: number;
+  year: number;
+}
+
+export interface CalendarDateValues{
+  day: number;
+  month: number;
+  year: number;
+  isCurrentMonth: boolean;
+}
+
+/**
+ * Date Calendar
+ * @slot - Place element content here
+ */
+
+export class DayChangeEvent extends ComposedEvent {
+  constructor(public event: Event, public day: number, public month: number, public year: number) {
+    super('daySelected');
+  }
+}
+
+export class FocusChangeEvent extends ComposedEvent {
+  constructor(public dateToBeFocusedRef: HTMLButtonElement | undefined) {
+    super('setDateFocus');
+  }
+}
+
+export class KeyboardNavigationFocusEvent extends ComposedEvent {
+  constructor(public event: Event, public day: number, public month: number, public year: number) {
+    super('onCalendarKeydown');
+  }
+}
+
+/**
+ * Calendar
+ * @slot - Place element content here
+ */
+@customElement('pf-calendar')
+export class PfCalendar extends LitElement {
+  static readonly styles = [styles];
+
+  private currentDate: Date = new Date();
+  private weekdays: number[] = [0, 1, 2, 3, 4, 5, 6]; // S, M, T, W, T, F, S
+  private weeks: number[] = [0, 1, 2, 3, 4, 5]; // 1 previous month week, 4 current month weeks, 1 next month week
+
+  // Input properties from the parent
+  @property() currentYear: number = this.currentDate.getFullYear();
```

Comment:
I noticed this isn't a public prop on pfv4 DatePicker or CalendarFormat, so do we need it to be a public `@property` here? or can it be a private field?

if it does need to be private, see notes on the demo file regarding attribute names
```suggestion
  @property({ type: Number, attribute: 'current-year' }) currentYear: number = this.currentDate.getFullYear();
```